### PR TITLE
feat(locale): add Senegal locale

### PR DESCRIPTION
**Pull request by makhtar-sarr**
**Description**

This pull request adds a new locale for Senegal to Faker-js. It allows Faker-js users to generate Senegal-specific data, including names, addresses and phone numbers.

**Changes made**

1. Creation of a local Senegal folder (`fr_SN`) in the appropriate directory.
2. Set up Senegal-specific data, including names, addresses and phone numbers.
3. Run `pnpm run preflight`.

**History**

Senegal is a West African country with its own culture and standards. Adding this locale will enable Faker-js users to generate data more relevant to Senegal, which can be useful for testing, generating demo data, etc.

**Expected result**

Faker-js users will be able to select the "senegal" locale and generate data that conforms to Senegalese culture and standards using Faker-js functions.

I will be available to answer your questions or make additional modifications if necessary.

### CODE REVIEW

**Comment by ST-DDT** [comment]
Hi @makhtar-sarr, thanks for your contribution.
Could you please do us a favor and split this PR into multiple smaller ones?
We prefer to review them at 1 PR per module (e.g. person/location), as checking foreign languages tends to take quite a while.
That way the first data can be merged without waiting on others. Please also include a description with wiki links or others sources for us to verify the entries (such as phone number patterns).
The first PR may contain both the metadata and the first module/locale category data (as well as the generated files).

See also:

- https://github.com/faker-js/faker/blob/next/CONTRIBUTING.md#adding-new-locale-or-updating-existing-one

File: src/locales/fr_SN/internet/free_email.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export default ['gmail.com', 'yahoo.ca', 'hotmail.com'];
```
Comment (matthewmayer) [inline src/locales/fr_SN/internet/free_email.ts]: Yahoo.ca is for Canada. Maybe yahoo.com ? Are there any local email providers popular in Senegal also?

**Review by matthewmayer — CHANGES_REQUESTED** [review]

**Comment by makhtar-sarr** [comment]
Hello @ST-DDT,

Thank you for your feedback and constructive comments. I divided the PR into several small PR, in this first PR are the metadata and the internet module. As for the wiki links, I preferred to put them in this comment. Otherwise, if they're to appear as comments in the modules, I can do so.

Wiki links: 
[metadata](https://en.wikipedia.org/wiki/List_of_countries_and_dependencies_and_their_capitals_in_native_languages#S)

Thanks again for your responsiveness and your advice.

File: src/locales/fr_SN/internet/free_email.ts (hunk 1 above)
Comment (makhtar-sarr) [reply src/locales/fr_SN/internet/free_email.ts]: My mistake, it's yohoo.com. Thank you for your comment

**Review by matthewmayer — DISMISSED** [review]

**Comment by matthewmayer** [comment]
The failing tests are a known issue with the `fr` locale (see #2503 ) - to work around this, you need to add fr_SN to lines 46, 47 and 48 of `test/all-functional.spec.ts`

**Review by ejcheng — DISMISSED** [review]

**Comment by ST-DDT** [comment]
If you need help with the test fixes please let us know.

**Comment by makhtar-sarr** [comment]
I modified lines 46, 47, and 48 of `test/all-functional.spec.ts` by adding `fr_SN`.

**Review by matthewmayer — APPROVED** [review]

**Review by ST-DDT — APPROVED** [review]

**Pull request merged**